### PR TITLE
Bling and StickRumble fixes and additions

### PR DESCRIPTION
**Pull request by KyleRAnderson**
Optimized the bling susbsytem for battery displaying, added a battery level calculator (very approximate). StickRumble command also has an added interval feature and works as it should now. 

*All has been tested on plyboy. Bling subsystem will not work on Mergius, therefore meaning the battery indications must be turned off.*

### CODE REVIEW

File: src/org/usfirst/frc/team2706/robot/RobotMap.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/RobotMap.java]: This isn't really how RobotMap is meant to be used.
The RobotMap constants end before `ROBOT_ID_LOC`
```Java
    // Bling Various
    // Number of pixels in LED Strip
    private static final int[] LED_STRIP_PIXELCOUNT_VALS = {120, 120, 120};
    public static final int LED_STRIP_PIXELCOUNT = getConstant("LED_STRIP_PIXELCOUNT ");
   
    // Battery Variable which equals the voltage of the battery
    private static final double[] BATTERY_CAPACITY_VALS = {12.0, 12.0, 12.0};
    public static final double BATTERY_CAPACITY = getConstant("BATTERY_CAPACITY ");
```

File: src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java]: Try spacing these out to make them more readable, or add comments

File: src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java]: Try to remove this if you can, if not just put the brackets like `{}`

File: src/org/usfirst/frc/team2706/robot/controls/StickRumble.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/controls/StickRumble.java]: Try to put comments above code to make it more readable

File: src/org/usfirst/frc/team2706/robot/controls/StickRumble.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/controls/StickRumble.java]: You don't need a : there

File: src/org/usfirst/frc/team2706/robot/subsystems/Bling.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/subsystems/Bling.java]: Try to put the comment above the line

File: src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java]: It would probably be easier to read if the comment was on the line above.

File: src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java]: Will this spam the RIOLog? That should probably be avoided if it does. Maybe print it every 5 minutes.

File: src/org/usfirst/frc/team2706/robot/controls/StickRumble.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/controls/StickRumble.java]: Try to put the comment above.

File: src/org/usfirst/frc/team2706/robot/controls/StickRumble.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/controls/StickRumble.java]: This is called even when `isFinished()` returns true, so it isn't really ended forcefully

**Review by ryanlarkin — DISMISSED** [review]
There are a couple of small issues with this pull request.
~1. How often are you printing to the RIO Log, we should only print sometimes, and if it is something you want to get the attention of the driver with, you should use DriverStation.warning() or something~
~2. Try to use RobotMap like the rest of the constants, and follow proper formatting for constants `LIKE_THIS`.~
3. There are some misc. formatting errors. (I can fix this up) (Made PR in your fork)
4. There are a lot of unnecessary empty lines. (I can fix this up) (Made PR in your fork)

File: src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java
Comment (KyleRAnderson) [reply src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java]: No, it wouldn't because batCritical turns to true the first time this occurs so it wouldn't run a second time. Do you actually want this or do you want a more prominent warning? I could even set a rumble on the controllers.

Comment (ryanlarkin) [reply src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java]: Maybe something like `DriverStation.getInstance().reportWarning("Low Battery " + batCritical", false);`. That would print directly to the driver station

File: src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java
Comment (cnnradams) [inline src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java]: There should be a space right before the comment line so we know which line its referring to and it looks nicer

Comment (KyleRAnderson) [reply src/org/usfirst/frc/team2706/robot/commands/BatteryReader.java]: Completed this in 8fcbc81aa2feffc00c20925578919f871f3e0090.

File: src/org/usfirst/frc/team2706/robot/RobotMap.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/RobotMap.java]: To use RobotMap properly, this should go with the rest of the constants before `ROBOT_ID_LOC` and look like:
```Java
    // Battery Variable which equals the voltage of the battery
    private static final double[] BATTERY_CAPACITY_VALS = {12.0, 12.0, 12.0};
    public static final double BATTERY_CAPACITY = getConstant("BATTERY_CAPACITY");
```

**Review by ryanlarkin — DISMISSED** [review]
Just need to fix RobotMap, and I can fix any minor formatting issues.

File: src/org/usfirst/frc/team2706/robot/RobotMap.java
Comment (KyleRAnderson) [reply src/org/usfirst/frc/team2706/robot/RobotMap.java]: Just got rid of that since it wasn't necessary for it to be there.

**Comment by KyleRAnderson** [comment]
Need to wait on #48 to have the new features outlined.

File: .classpath
Comment (ryanlarkin) [inline .classpath]: These shouldn't be removed, otherwise the talons and NavX will stop working

**Review by ryanlarkin — DISMISSED** [review]
The NavX and CAN Talon libraries need to be re-added to the classpath. Also I have a pull request open to fix some formatting stuff in your fork.

File: .classpath
Comment (KyleRAnderson) [reply .classpath]: Eclipse did it automatically. I don't seem to have the proper modules for them installed. How can I get them installed?

Comment (KyleRAnderson) [reply .classpath]: Nevermind. Figured it out.

**Comment by ryanlarkin** [comment]
Just deploy the code, you don't need to be connected to the robot, then
restart eclipse

File: src/org/usfirst/frc/team2706/robot/subsystems/GearHandler.java
Comment (cnnradams) [inline src/org/usfirst/frc/team2706/robot/subsystems/GearHandler.java]: No spaces here

**Review by cnnradams — APPROVED** [review]
I'm happy with it maybe wait for Ryan if he wants to look a little deeper? Nice documentation by the way 👍

File: .classpath
Comment (ryanlarkin) [inline .classpath]: This will not deploy correctly, it needs to be exactly the same as it was before.

**Review by ryanlarkin — CHANGES_REQUESTED** [review]
The .classpath still isn't correct, also could you take a look at https://github.com/eAUE/2017-2706-Robot/pull/1

File: src/org/usfirst/frc/team2706/robot/subsystems/GearHandler.java
Comment (KyleRAnderson) [reply src/org/usfirst/frc/team2706/robot/subsystems/GearHandler.java]: Done

File: .classpath
Comment (KyleRAnderson) [reply .classpath]: Sorry. Fixed it in 35e910c

**Review by ryanlarkin — CHANGES_REQUESTED** [review]
Instead of adding a teleopInit, auto, teleopPeriodic, clear to Bling, make a command that can do this. You can make it the default command. You can use DriverStation.isTelop() if you want too.

**Comment by KyleRAnderson** [comment]
I made a seperate command to replace teleopPeriodic, but clear, teleopInit and auto are all only run once so I didn't think it was worth it.

**Comment by ryanlarkin** [comment]
If possible, I'd still want to keep Robot as minimal as possible, so I made a small change to the command, not sure if it works though. eAUE#2

File: src/org/usfirst/frc/team2706/robot/Robot.java
Comment (ryanlarkin) [inline src/org/usfirst/frc/team2706/robot/Robot.java]: In the Bling command, when it is canceled or ended, it should cancel automatically.

Comment (KyleRAnderson) [reply src/org/usfirst/frc/team2706/robot/Robot.java]: That command doesn't cancel, it tells the arduino to clear the LED strip. Otherwise, the pattern will stay on. I guess we could leave it to the command to do that in the `end()` and `interrupted()` commands. I'll change that. Done in f0f2e88.

**Review by ryanlarkin — APPROVED** [review]
Looks good

**Comment by ryanlarkin** [comment]
@eAUE, is this ready to be merged, or does it need testing?

**Comment by KyleRAnderson** [comment]
It needs to be tested. I'm going to be at the KRP to do that today.

**Comment by KyleRAnderson** [comment]
@eandr127 @Stealth2994 can you guys put the new robot code with the new controller mappings on GitHub by the way for drive practise later today? Thanks!

**Comment by ryanlarkin** [comment]
It's currently in GearWallAlignment branch, I can hopefully have it merged
into master by 1.

**Comment by KyleRAnderson** [comment]
@eandr127 @Stealth2994 so all of my code is working now, but I have the new stuff from the practice branch in here as well. I'll wait for you guys to merge that before I merge this.

File: src/org/usfirst/frc/team2706/robot/Robot.java
Comment (KyleRAnderson) [inline src/org/usfirst/frc/team2706/robot/Robot.java]: I'm going to get the `BlingPeriodic` command doing this, no need to get the `autonomousInit` function doing it.

**Comment by cnnradams** [comment]
I think this is ready to be merged... lol :)

**Comment by KyleRAnderson** [comment]
Yeah, just before we do that how accurate is the IR sensor at detecting when the peg is in the hole?

**Comment by KyleRAnderson** [comment]
I guess it doesn't really matter. I'll merge it.

**Pull request merged**
